java/unittests/infodynamics/measures/continuous/gaussian/SInfoCalculatorGaussianTester.java

```diff
@@ -0,0 +1,129 @@
+ *  
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package infodynamics.measures.continuous.gaussian;
+
+import infodynamics.utils.MatrixUtils;
+import infodynamics.utils.RandomGenerator;
+import junit.framework.TestCase;
+
+public class SInfoCalculatorGaussianTester extends TestCase {
+
+  /**
+   * For two variables, S-info is twice the MI between them
+   */
+  public void testTwoVariables() throws Exception {
+    double[][] cov = new double[][] {{1, 0.5}, {0.5, 1}};
+
+    SInfoCalculatorGaussian sCalc = new SInfoCalculatorGaussian();
+    sCalc.initialise(2);
+    sCalc.setCovariance(cov);
+    double sinfo = sCalc.computeAverageLocalOfObservations();
+
+    MutualInfoCalculatorMultiVariateGaussian miCalc = new MutualInfoCalculatorMultiVariateGaussian();
+    miCalc.initialise(1,1);
+    miCalc.setCovariance(cov, false);
+    double mi = miCalc.computeAverageLocalOfObservations();
+
+    assertEquals(sinfo, 2*mi, 1e-6);
+  }
+
+
+  /**
+   * Compare against the direct calculation of S-info as a sum of mutual informations.
+   */
+  public void testCompareWithEntropy() throws Exception {
+    double[][] cov = new double[][] {{1, 0.4, 0.3}, {0.4, 1, 0.2}, {0.3, 0.2, 1}};
+
+    SInfoCalculatorGaussian sCalc = new SInfoCalculatorGaussian();
+    sCalc.initialise(3);
+    sCalc.setCovariance(cov);
+    double sinfo = sCalc.computeAverageLocalOfObservations();
+
+    // Calculate using a mutual info calculator and picking submatrices manually
+    MutualInfoCalculatorMultiVariateGaussian miCalc = new MutualInfoCalculatorMultiVariateGaussian();
+    miCalc.initialise(2,1);
+    miCalc.setCovariance(cov, false);
+    double sinfo_miCalc = miCalc.computeAverageLocalOfObservations();
+
+    miCalc.initialise(2,1);
+    miCalc.setCovariance(MatrixUtils.selectRowsAndColumns(cov, new int[] {2,0,1}, new int[] {2,0,1}), false);
+    sinfo_miCalc += miCalc.computeAverageLocalOfObservations();
+    miCalc.initialise(2,1);
+    miCalc.setCovariance(MatrixUtils.selectRowsAndColumns(cov, new int[] {1,2,0}, new int[] {1,2,0}), false);
+    sinfo_miCalc += miCalc.computeAverageLocalOfObservations();
+
+    assertEquals(sinfo, sinfo_miCalc, 1e-6);
+
+  }
+
+  /**
+   * Confirm that the local values average correctly back to the average value
+   */
+  public void testLocalsAverageCorrectly() throws Exception {
+
+    int dimensions = 4;
+    int timeSteps = 1000;
+    SInfoCalculatorGaussian sCalc = new SInfoCalculatorGaussian();
+    sCalc.initialise(dimensions);
+
+    // generate some random data
+    RandomGenerator rg = new RandomGenerator();
+    double[][] data = rg.generateNormalData(timeSteps, dimensions,
+        0, 1);
+
+    sCalc.setObservations(data);
+
+    double sinfo = sCalc.computeAverageLocalOfObservations();
+    double[] sLocal = sCalc.computeLocalOfPreviousObservations();
+
+    System.out.printf("Average was %.5f\n", sinfo);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.